scRNA-seq-advanced/04-differential_expression.Rmd

```diff
@@ -12,156 +12,255 @@ output:
+The final `SingleCellExperiment`(SCE) object, stored in `data/rms/integrated/rms_all_sce.rds`.
+
+We also have provided a metadata file, `data/rms/annotations/rms_sample_metadata.tsv`, that contains information from each sample, such as diagnosis, sex, age, etc.
+In this file each row corresponds to a sample found in the integrated SCE object.
+
+To begin, let's set up our directories and files:
+
+```{r filepaths}
 # set up file paths 
 # data directory for RMS data
 data_dir <- file.path("data", "rms")
 
-# integrated file containing samples to for DE 
+# integrated file containing all libraries to use for DE analysis
 integrated_sce_file <- file.path(data_dir, "integrated", "rms_all_sce.rds")
 
-# sample metadata with information needed to setup DE
+# sample metadata with information needed to setup DE analysis
 sample_metadata_file <- file.path(data_dir, "annotations", "rms_sample_metadata.tsv")
 
-# output files 
+# directory to store output
 deseq_dir <- file.path("analysis", "rms", "deseq")
 if(!dir.exists(deseq_dir)){
   dir.create(deseq_dir, recursive = TRUE)
 }
 
-# results from deseq
+# results file to output from DE analysis
 deseq_output_file <- file.path(deseq_dir, "rms_myoblast_deseq_results.tsv")
 ```
 
+We can go ahead and read in the SCE object and the metadata file.
 
-```{r}
-# read in integrated SCE object 
+```{r read files}
+# read in the SCE object that has already been integrated
 integrated_sce <- readr::read_rds(integrated_sce_file)
 
 # read in sample metadata file 
 sample_metadata <- readr::read_tsv(sample_metadata_file)
 ```
 
-Before we set up our experiment, let's explore our dataset a little.
+## Dataset Exploration
 
-```{r}
-# First just plot by sample
+Before we dive into differential expression, let's explore our integrated SCE object and the dataset a little more.
+
+We'll start by looking at what's inside the object.
+Here we should have data from both the uncorrected and the corrected, integrated data for both the gene expression data and the reduced dimensionality results.
+How are those stored in our object?
+
+```{r dimnames }
+# look at the assay names in our object
+assayNames(integrated_sce)
+
+# look at the names of the dimensionality reduction present in the SCE object
+reducedDimNames(integrated_sce)
+```
+
+When we look at the assay names we should see that there are 3 matrices, `counts`, `logcounts`, and `fastmnn_corrected`. 
+The `counts` and `logcounts` assay correspond to the uncorrected gene expression data that has just been merged and has NOT been integrated.
+The `fastmnn_corrected` data contains the corrected gene expression data obtained from integration. 
+For this exercise we will not be using the `fastmnn_corrected` data (more on why not once we get to setting up the differential expression), but we need to be aware that it is present and be able to distinguish it from our uncorrected data. 
```

Comment:
I think we might be able to make a plot (like the violin) plot we have at the end with the corrected data instead of `logcounts` and show that you get negative gene expression values? I don't know if that's really necessary though.